library/std/src/collections/mod.rs

```diff
@@ -79,34 +79,45 @@
-//! operations, the collection's size is denoted by n. If another collection is
-//! involved in the operation, it contains m elements. Operations which have an
-//! *amortized* cost are suffixed with a `*`. Operations with an *expected*
-//! cost are suffixed with a `~`.
-//!
-//! All amortized costs are for the potential need to resize when capacity is
-//! exhausted. If a resize occurs it will take *O*(*n*) time. Our collections never
-//! automatically shrink, so removal operations aren't amortized. Over a
-//! sufficiently large series of operations, the average cost per operation will
-//! deterministically equal the given cost.
-//!
-//! Only [`HashMap`] has expected costs, due to the probabilistic nature of hashing.
-//! It is theoretically possible, though very unlikely, for [`HashMap`] to
-//! experience worse performance.
-//!
-//! ## Sequences
-//!
-//! |                | get(i)                 | insert(i)               | remove(i)              | append    | split_off(i)           |
-//! |----------------|------------------------|-------------------------|------------------------|-----------|------------------------|
-//! | [`Vec`]        | *O*(1)                 | *O*(*n*-*i*)*           | *O*(*n*-*i*)           | *O*(*m*)* | *O*(*n*-*i*)           |
-//! | [`VecDeque`]   | *O*(1)                 | *O*(min(*i*, *n*-*i*))* | *O*(min(*i*, *n*-*i*)) | *O*(*m*)* | *O*(min(*i*, *n*-*i*)) |
-//! | [`LinkedList`] | *O*(min(*i*, *n*-*i*)) | *O*(min(*i*, *n*-*i*))  | *O*(min(*i*, *n*-*i*)) | *O*(1)    | *O*(min(*i*, *n*-*i*)) |
+//! Throughout the documentation, we will adhere to the following conventions for
+//! operation notation:
+//!
+//! * The collection's size is denoted by `n`.
+//! * If a second collection is involved, it's size is denoted by `m`.
+//! * Item indices are denoted by `i`.
+//! * Operations which have an *amortized* cost are suffixed with a `*`.
+//! * Operations with an *expected* cost are suffixed with a `~`.
+//!
+//! Calling operations that add to a collection will occasionally require a
+//! collection to be resized - an extra operation that takes *O*(*n*) time.
+//!
+//! *Amortized* costs are calculated to account for the time cost of such resize
+//! operations *over a sufficiently large series of operations*. An individual
+//! operation may be slower or faster due to the sporadic nature of collection
+//! resizing, however the average cost per operation will eventually equal the
+//! amortized cost.
+//!
+//! Rust's collections never automatically shrink, so removal operations aren't
+//! amortized.
+//!
+//! [`HashMap`] uses *expected* costs to denote that it is theoretically possible,
+//! though very unlikely, for [`HashMap`] to experience worse performance than
+//! the expected cost. This is due to the probablilistic nature of hashing -
+//! i.e. it is possible to generate a duplicate hash given some input key that
+//! requires extra computation to correct, but exceptionally unlikely.
```

Comment:
The [birthday paradox](https://en.wikipedia.org/wiki/Birthday_problem) actually essentially guarantees this for smaller maps once they reach over about half their fill, because the hash's size must still be fit into the map's allocation, and we use open addressing. If memory serves, we reallocate early to reduce this issue, but still, it seems incorrect to say "*exceptionally* unlikely".

Reply:
Lmaaaoo my javascript brain took over and I didn't think about allocation size

Reply:
tbh it probably also affects JS maps! ...it just doesn't matter for them.